### PR TITLE
 [BP-3.1.1][FLINK-35149][cdc-composer] Fix DataSinkTranslator#sinkTo ignoring pre-write topology if not TwoPhaseCommittingSink (#3233)

### PR DESCRIPTION
**Pull request by loserwang1024**
BP https://issues.apache.org/jira/browse/FLINK-35149 to 3.1.1

### CODE REVIEW

**Comment by loserwang1024** [comment]
@PatrickRen , CC

File: flink-cdc-composer/src/test/java/org/apache/flink/cdc/composer/flink/translator/DataSinkTranslatorTest.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+import org.apache.flink.runtime.jobgraph.OperatorID;
+import org.apache.flink.streaming.api.connector.sink2.WithPreWriteTopology;
+import org.apache.flink.streaming.api.datastream.DataStream;
+import org.apache.flink.streaming.api.datastream.DataStreamSource;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.apache.flink.streaming.api.transformations.OneInputTransformation;
+
+import org.apache.flink.shaded.guava31.com.google.common.collect.Lists;
+
+import org.junit.Assert;
```
Comment (Jiabao-Sun) [inline flink-cdc-composer/src/test/java/org/apache/flink/cdc/composer/flink/translator/DataSinkTranslatorTest.java:32]: Mixed using junit4 and junit5.

File: flink-cdc-composer/src/test/java/org/apache/flink/cdc/composer/flink/translator/DataSinkTranslatorTest.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+/** A test for {@link DataSinkTranslator}. */
+public class DataSinkTranslatorTest {
+
+    @Test
+    public void testPreWriteWithoutCommitSink() {
```
Comment (Jiabao-Sun) [inline flink-cdc-composer/src/test/java/org/apache/flink/cdc/composer/flink/translator/DataSinkTranslatorTest.java:42]:
```suggestion
class DataSinkTranslatorTest {

    @Test
    void testPreWriteWithoutCommitSink() {
```

File: flink-cdc-composer/src/test/java/org/apache/flink/cdc/composer/flink/translator/DataSinkTranslatorTest.java (hunk 1 above)
Comment (loserwang1024) [reply flink-cdc-composer/src/test/java/org/apache/flink/cdc/composer/flink/translator/DataSinkTranslatorTest.java]: Thanks, i will also modified it in master

**Review by Jiabao-Sun — APPROVED** [review]
Thanks @loserwang1024 for the quick fix.
LGTM.

**Pull request merged**